hail/Makefile

```diff
@@ -41,13 +42,25 @@ SCALA_BUILD_INFO := src/main/resources/build-info.properties
 
 SHADOW_JAR := build/libs/hail-all-spark.jar
 SHADOW_TEST_JAR := build/libs/hail-all-spark-test.jar
+DEBUG_CLASSES := build/debug
 PYTHON_JAR := python/hail/backend/hail-all-spark.jar
 WHEEL := build/deploy/dist/hail-$(HAIL_PIP_VERSION)-py3-none-any.whl
 EGG := build/deploy/dist/hail-$(HAIL_PIP_VERSION)-py3.6.egg
 
 .PHONY: shadowJar
+ifdef HAIL_DEBUG_MODE
+shadowJar: debugShadowJar
+endif
 shadowJar: $(SHADOW_JAR)
 
+.PHONY: debugShadowJar
+debugShadowJar: $(SHADOW_JAR)
+	mkdir -p build/debug/scala/is/hail/annotations/ && \
+	javac src/debug/scala/is/hail/annotations/Memory.java && \
+	mv src/debug/scala/is/hail/annotations/Memory.class build/debug/scala/is/hail/annotations && \
```

Comment:
the ampersands and `\` are unnecessary, make will fail if any step fails

Reply:
I think these three lines can be:
```
mkdir -p build/debug/scala/is/hail/annotations
javac -d build/debug src/debug/scala/is/hail/annotations/Memory.java
```

Reply:
What Dan said. Also prefix your `mkdir` commands with `@` to suppress echoing of the commands by `make`.